cirq-core/cirq/transformers/eject_phased_paulis_test.py

```diff
@@ -0,0 +1,590 @@
+            [cirq.X(a)],
+        ),
+        eject_parameterized=True,
+    )
+
+    # Double cross.
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=0.125).on(a)],
+            [cirq.PhasedXPowGate(phase_exponent=0.375).on(b)],
+            [cirq.CZ(a, b) ** 0.25],
+        ),
+        expected=quick_circuit(
+            [cirq.CZ(a, b) ** 0.25],
+            [
+                cirq.PhasedXPowGate(phase_exponent=0.5).on(b),
+                cirq.PhasedXPowGate(phase_exponent=0.25).on(a),
+            ],
+        ),
+    )
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=x).on(a)],
+            [cirq.PhasedXPowGate(phase_exponent=y).on(b)],
+            [cirq.CZ(a, b) ** z],
+        ),
+        expected=quick_circuit(
+            [cirq.CZ(a, b) ** z],
+            [
+                cirq.PhasedXPowGate(phase_exponent=y + z / 2).on(b),
+                cirq.PhasedXPowGate(phase_exponent=x + z / 2).on(a),
+            ],
+        ),
+        eject_parameterized=True,
+    )
+
+
+def test_toggles_measurements():
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+    x = sympy.Symbol('x')
+
+    # Single.
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=0.25).on(a)],
+            [cirq.measure(a, b)],
+        ),
+        expected=quick_circuit(
+            [cirq.measure(a, b, invert_mask=(True,))],
+        ),
+    )
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=0.25).on(b)],
+            [cirq.measure(a, b)],
+        ),
+        expected=quick_circuit(
+            [cirq.measure(a, b, invert_mask=(False, True))],
+        ),
+    )
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=x).on(b)],
+            [cirq.measure(a, b)],
+        ),
+        expected=quick_circuit(
+            [cirq.measure(a, b, invert_mask=(False, True))],
+        ),
+        eject_parameterized=True,
+    )
+
+    # Multiple.
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=0.25).on(a)],
+            [cirq.PhasedXPowGate(phase_exponent=0.25).on(b)],
+            [cirq.measure(a, b)],
+        ),
+        expected=quick_circuit(
+            [cirq.measure(a, b, invert_mask=(True, True))],
+        ),
+    )
+
+    # Xmon.
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=0.25).on(a)],
+            [cirq.measure(a, b, key='t')],
+        ),
+        expected=quick_circuit(
+            [cirq.measure(a, b, invert_mask=(True,), key='t')],
+        ),
+    )
+
+    # CCOs
+    assert_optimizes(
+        before=quick_circuit(
+            [cirq.PhasedXPowGate(phase_exponent=0.25).on(a)],
+            [cirq.measure(a, key="m")],
+            [cirq.X(b).with_classical_controls("m")],
```

Comment:
What would be the expected output for something like ?
```python3
[cirq.PhasedXPowGate(phase_exponent=0.25).on(a)],
[cirq.measure(a, key="m")],
[cirq.X(b).with_classical_controls("m")],
[cirq.measure(b, key='m2')]
```
?

Reply:
`cirq.X(b).with_classical_controls` will be unaffected because it's not a unitary operation. So, the output will be:
```
a: ───!M─────────────────
      ║
b: ───╫────X───M('m2')───
      ║    ║
m: ═══@════^═════════════
```